### ISSUE TITLE
require-return-type says "must annotate undefined return type" on an async generator function

### ISSUE THREAD

**Issue by jedwards1211** [gajus/eslint-plugin-flowtype#386]
```js
const resolvers = {
  TagState: {
    async *subscribe(): AsyncIterable<TagState> {
      await ...
      yield {...}
    }
  }
}
```

```
error  Must annotate undefined return type  flowtype/require-return-type
```

Annotating `: void` of course makes flow complain about the return type.

```
    "flowtype/require-return-type": [
      2,
      "always",
      {
        "annotateUndefined": "always",
        "excludeArrowFunctions": true,
        "excludeMatching":[
          "constructor", "componentDidMount",
          "componentDidUpdate", "componentWillReceiveProps",
          "componentWillUnmount"
        ]
      }
    ],
```

eslint: 5.13.0
eslint-plugin-flowtype: 3.4.1

**Comment by gajus** [comment]
Thank you

**Comment by jedwards1211** [comment]
Thanks!  It was really easy to figure out where to fix this and how to write tests for it.  Good job on making the code organized! 💯
